src/pathfinder/yapf/yapf_ship.cpp

```diff
@@ -172,28 +172,28 @@ class CYapfFollowShipT
 		return 'w';
 	}
 
-	/** Returns a random trackdir that can be reached from the current tile/trackdir, or INVALID_TRACK if none is available. */
-	static Trackdir GetRandomFollowUpTrackdir(const Ship *v, TileIndex tile, Trackdir dir, bool include_90_degree_turns)
+	/** Returns a random tile/trackdir that can be reached from the current tile/trackdir, or tile/INVALID_TRACK if none is available. */
+	static std::pair<TileIndex, Trackdir> GetRandomFollowUpTrackdir(const Ship *v, TileIndex tile, Trackdir dir, bool include_90_degree_turns)
 	{
 		TrackFollower follower(v);
 		if (follower.Follow(tile, dir)) {
-			tile = follower.m_new_tile;
 			TrackdirBits dirs = follower.m_new_td_bits;
 			if (!include_90_degree_turns) dirs &= ~TrackdirCrossesTrackdirs(dir);
 			const int strip_amount = _random.Next(CountBits(dirs));
 			for (int s = 0; s < strip_amount; ++s) RemoveFirstTrackdir(&dirs);
-			return FindFirstTrackdir(dirs);
+			return std::make_pair(follower.m_new_tile, FindFirstTrackdir(dirs));
```

Comment:
It's not wrong to use std::make_pair, but you can also initialize a pair using braced initialization: `{follower.m_new_tile, FindFirstTrackdir(dirs)}`